src/Database/LSMTree/Internal/IndexCompact.hs

```diff
@@ -618,44 +615,50 @@ indexVersion :: Word32
 indexVersion = 1
 
 -- | 32 bits, to be used before writing any other parts of the serialised file!
-headerBuilder :: BB.Builder
-headerBuilder = BB.word32Host indexVersion
+headerLBS :: LBS.ByteString
+headerLBS =
+    -- create a single 4 byte chunk
+    BB.toLazyByteStringWith (BB.safeStrategy 4 BB.smallChunkSize) mempty $
+      BB.word32Host indexVersion
 
 -- | A chunk of the primary array, which can be constructed incrementally.
 data Chunk = Chunk { cPrimary :: !(VU.Vector Word32) }
   deriving stock (Show, Eq)
 
 -- | 32 bit aligned.
-chunkBuilder :: Chunk -> BB.Builder
-chunkBuilder Chunk {..} = putVec32 cPrimary
+chunkToBS :: Chunk -> BS.ByteString
+chunkToBS (Chunk (VU.V_Word32 (PV.Vector off len ba))) =
+    byteArrayToByteString (mul4 off) (mul4 (off + len)) ba
```

Comment:
This doesn't look right. The `byteArrayToByteString` takes an offset and a length, not two offsets, so surely it should be:
```suggestion
    byteArrayToByteString (mul4 off) (mul4 len) ba
```
Assuming this was wrong, why didn't the tests catch it? Perhaps the offset was always 0?

Reply:
Good catch. This must have snuck in because of `byteArrayFromTo` taking two offsets.

Indeed, all chunks start at offset 0. They usually span the whole byte array, with the exception of the last chunk, where the unused end is sliced off.